### ISSUE TITLE
Evaluate expressions during debugging.

### ISSUE THREAD

**Issue by megri** [scalameta/metals#3051]
**Describe the solution you'd like**
While debugging there is a debug console. Any command entered into it currently results in the message "an implementation is missing". It would be nice if the console had access to the currently viewed stack-frame and allowed evaluation of arbitrary expressions.

**Search terms**:
debug, debugger, evaluate expression

**Comment by adpi2** [comment]
@tgodzik should we close this issue since #2959 got merged?

**Comment by tgodzik** [comment]
We still need Scala 3 support to close it or we can raise another issue.

**Comment by sa-** [comment]
Cool to see this merged!
I still see `an implementation is missing` when I try to debug, is it going to be a while until this is included in the plugin?

**Comment by tgodzik** [comment]
You should be able to update to newest snapshot of metals and restart the build server to test it out. We are still working on some fixes for a smoother experience for the users, but it would be cool to hear feedback already!

**Comment by sa-** [comment]
I'm new to VS Code and language servers, what's the easiest way for me to install a snapshot build?

**Comment by sa-** [comment]
Never mind, I found this: https://scalameta.org/metals/docs/editors/vscode#using-latest-metals-snapshot

Thanks again for working on this!

**Comment by sa-** [comment]
It looks like there is some autocomplete when I try to type in the console, but it still says an implementation is missing when I try to run this.

![image](https://user-images.githubusercontent.com/1297369/135877970-d3f3b164-8cf2-4c55-ab6e-e89213b0d7a1.png)

I am running this in an instance of VS Code that is attached to a pod in kubernetes, if this makes a difference.

**Comment by sa-** [comment]
`Metals version: 0.10.7+78-03c450a0-SNAPSHOT`

`Connected to Build server: Bloop v1.4.9-18-1ef3bf32`

Today the error is a bit different: 

```
> indexVendors.count()
Cannot evaluate because of ch.epfl.scala.debugadapter.internal.evaluator.ExpressionCompilationFailed: .
```

**Comment by adpi2** [comment]
@sa- That looks like a bug. Expression evaluation in Metals is provided by https://github.com/scalacenter/scala-debug-adapter. Could you find a minimization of the bug and report it there? Thanks a lot.

**Comment by sa-** [comment]
Do you mean steps to reproduce?

**Comment by adpi2** [comment]
Yes, we need the piece of code, the position of the breakpoint and the expression to reproduce the bug.

**Comment by sa-** [comment]
Sure. 

- Run this container: https://gist.github.com/sa-/09c8a347d0dbe864d29d1f8f66eb8e88
- Attach VS Code to the running container
- Install metals and set the server version to `0.10.7+78-03c450a0-SNAPSHOT`
- create a new scala project using the gitter 8 `hello-world` template
- Import build
- add a breakpoint and hit debug above `object Main extends App`
- Try to evaluate expression unsuccessfully :(

![image](https://user-images.githubusercontent.com/1297369/135992561-49ff9d87-649c-4902-9ec3-54eed1b10188.png)

**Comment by tgodzik** [comment]
That seems to be an issue with definitions, not related to your container. Will take a look!

**Comment by tgodzik** [comment]
Fixing it here https://github.com/scalacenter/scala-debug-adapter/pull/113

**Comment by Pardhu448** [comment]
Hi, I am also facing the same issue. _Metals version: _1.10.11__

![metals_debugissue](https://user-images.githubusercontent.com/10006246/136857520-ceed8970-1d1c-477e-96d7-0ab2492a3b2b.png)

**Comment by tgodzik** [comment]
@tdudzik could you check a look if this is fixed in the most recent version of Scala debug?

**Comment by Pardhu448** [comment]
@adpi2 I am using [MUnit](https://scalameta.org/munit/) testing library.

**Comment by adpi2** [comment]
I managed to reproduce the bug with the current snapshot of Metals. Also with the latest version of the debug adapter it seems the bug is fixed, but we have another bug now:

When I evaluate `1+1` the result is `"values are not the same"`. Curiously this is not an error. The debugger really thinks that this is the result of `1 + 1`. The root cause seems to be that there is a  by-name parameter called `clue` in `assertEquals` whose value is  `"values are not the same"`.

Reproduction test in https://github.com/scalacenter/scala-debug-adapter/pull/140.

**Comment by Tarek98** [comment]
For anyone reading this issue in the future, it looks like Expression Evaluation is only supported in Scala 2.12 and 2.13 for Metals on VS Code. I wasn't able to get it working with Scala 2.11.

The supporting docs also mention that version requirement: https://contributors.scala-lang.org/t/the-scala-debug-adapter/4959/2
